table/tables/cache_test.go

```diff
@@ -111,5 +113,51 @@ func TestCacheTableBasicScan(t *testing.T) {
 		tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1105 IndexMerge is inapplicable or disabled"))
 	}
 	assertSelect()
+}
+
+func TestCacheTableBasicReadAndWrite(t *testing.T) {
+	t.Parallel()
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk1 := testkit.NewTestKit(t, store)
+	tk.MustExec("drop table if exists write_tmp1")
+	tk.MustExec("create  table write_tmp1 (id int primary key auto_increment, u int unique, v int)")
+	tk.MustExec("insert into write_tmp1 values" +
+		"(1, 101, 1001), (3, 113, 1003)",
+	)
+	tk.MustExec("alter table write_tmp1 cache")
+	// Read and add read lock
+	tk.MustQuery("select *from write_tmp1").Check(testkit.Rows("1 101 1001",
+		"3 113 1003"))
 
+	time.Sleep(1)
+	// read lock should valid
+	result := tk.MustQuery("explain format = 'brief' select *from write_tmp1")
+	result.Check(testkit.Rows("UnionScan 10000.00 root  ",
```

Comment:
Check the UnionScan in the plan is not quite stable, since trigger the cache reload is asynchronous.
I see why you call time.Sleep above...

Reply:
okay